ionoscloud/resource_server.go

```diff
@@ -1529,23 +1542,43 @@ func setResourceServerData(ctx context.Context, client *ionoscloud.APIClient, d
 		return fmt.Errorf("server entities cannot be empty for %s", d.Id())
 	}
 
-	if server.Properties != nil && server.Properties.BootVolume != nil {
-		volume, apiResponse, err := client.ServersApi.DatacentersServersVolumesFindById(ctx, datacenterId, d.Id(), *server.Properties.BootVolume.Id).Execute()
-		logApiRequestTime(apiResponse)
-		if err == nil {
-			var volumes []interface{}
+	inlineVolumeIds := d.Get("inline_volume_ids")
+
+	if inlineVolumeIds != nil {
+		inlineVolumeIds := inlineVolumeIds.([]interface{})
+
+		var volumes []interface{}
+		for i, volumeId := range inlineVolumeIds {
+			volume, apiResponse, err := client.ServersApi.DatacentersServersVolumesFindById(ctx, datacenterId, d.Id(), volumeId.(string)).Execute()
+			logApiRequestTime(apiResponse)
+			if err != nil {
+				return fmt.Errorf("error retrieving inline volume %w", err)
+			}
+			volumePath := fmt.Sprintf("volume.%d.", i)
 			entry := SetVolumeProperties(volume)
-			userData := d.Get("volume.0.user_data")
+			userData := d.Get(volumePath + "user_data")
 			entry["user_data"] = userData
 			if *server.Properties.Type != constant.VCPUType {
-				entry["ssh_key_path"] = d.Get("volume.0.ssh_key_path")
+				entry["ssh_key_path"] = d.Get(volumePath + "ssh_key_path")
 			}
-			backupUnit := d.Get("volume.0.backup_unit_id")
+			backupUnit := d.Get(volumePath + "backup_unit_id")
 			entry["backup_unit_id"] = backupUnit
-			volumes = append(volumes, entry)
-			if err := d.Set("volume", volumes); err != nil {
-				return fmt.Errorf("error setting volume %w", err)
+			bootOrder := ""
```

Comment:
Can't you just set it to AUTO here and not have the if 			if bootOrder == "" below?

Reply:
has been reworked, we will now use a boolean instead of predefined strings like 'AUTO', 'NONE', 'PRIMARY'